src/sentry/models/auditlogentry.py

```diff
@@ -67,15 +68,81 @@ class Meta:
     __repr__ = sane_repr("organization_id", "type")
 
     def save(self, *args, **kwargs):
+        # trim label to the max length
+        self._apply_actor_label()
+        self.actor_label = self.actor_label[:MAX_ACTOR_LABEL_LENGTH]
+        super().save(*args, **kwargs)
+
+    def _apply_actor_label(self):
         if not self.actor_label:
             assert self.actor or self.actor_key
             if self.actor:
                 self.actor_label = self.actor.username
             else:
                 self.actor_label = self.actor_key.key
-        # trim label to the max length
-        self.actor_label = self.actor_label[:MAX_ACTOR_LABEL_LENGTH]
-        super().save(*args, **kwargs)
+
+    def save_or_write_to_kafka(self):
+        """
+        Customer Silos do not have access to the AuditLogEntry table which is specific to the control silo.
+        For those silos, this method publishes the attempted audit log write to a durable kafka queue synchronously
+        that will eventually be consumed by the control silo.  For the control silo, this method ultimately results
+        in a save() call.
+
+        This method is most ideal for shared code paths that may be invoked from either control or customer silos,
+        but is not recommended on code paths that should always be invoked from the control silo and depend on the
+        synchronous database access.
+        """
+        from sentry.region_to_control.producer import produce_audit_log_entry
+
+        produce_audit_log_entry(self)
```

Comment:
https://github.com/getsentry/sentry/pull/39890/files/56a959ca0fb4f22536b92b2152e2837d82540191#diff-ef727c66539b7f5b7f3f26bad4450915485058f99b0ef95d2e04fa15649226a9R25

The reason I put the split logic there was simply because the `_should_write_to_kafka` was private to the consumer module.  I'm indifferent if you would prefer that be extracted here instead.